src/Repository/TransactionRepository.php

```diff
@@ -44,4 +44,24 @@ public function findAllSuccessfulTransactionByEmail(string $emailAddress): array
             ->getResult()
         ;
     }
+
+    /**
+     * Total amount in cents
+     */
+    public function getTotalAmountCurrentYearByEmail(string $email): int
+    {
+        return (int) $this->createQueryBuilder('transaction')
+            ->innerJoin('transaction.donation', 'donation')
+            ->select('SUM(donation.amount)')
+            ->where('donation.emailAddress = :email')
+            ->andWhere('transaction.payboxResultCode = :success_code')
+            ->andWhere('YEAR(transaction.payboxDateTime) = YEAR(NOW())')
```

Comment:
Perhaps it's more reliable and/or consistent to use php server time instead of the database server time.